src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -24,6 +27,10 @@ public Task Start(IMessageSession session)
                 var task = startable.Start(session).ThrowIfNull();
 
                 var startable1 = startable;
+                var startableName = startable1.GetType().AssemblyQualifiedName;
+
+                var longRunningTokenSource = 
+                    LongRunningWarning($"The start method for {startableName} is taking a long time. This blocks endpoint startup.");
```

Comment:
@colin-higgins - How about something like:

`The start method for {startableName} is taking too long to complete. The endpoint will not start until this operation has completed. Until then messages that arrive in the endpoint's queue will not be processed.`